src/drivers/opengles/lv_opengles_egl.c

```diff
@@ -0,0 +1,557 @@
+}
+
+void * lv_opengles_egl_window_get_display(lv_opengles_window_t * window)
+{
+    LV_UNUSED(window);
+    return (void *)(uintptr_t)egl_display;
+}
+
+void lv_opengles_egl_window_set_user_data(lv_opengles_window_t * window, void * user_data)
+{
+    window->user_data = user_data;
+}
+
+void * lv_opengles_egl_window_get_user_data(lv_opengles_window_t * window)
+{
+    return window->user_data;
+}
+
+void lv_opengles_window_delete(lv_opengles_window_t * window)
+{
+    EGLBoolean res = eglDestroySurface(egl_display, window->surface);
+    if(res == EGL_FALSE) {
+        LV_LOG_ERROR("eglDestroySurface failed.");
+        return;
+    }
+
+    lv_opengles_window_texture_t * texture;
+    while((texture = lv_ll_get_head(&window->textures))) {
+        if(texture->texture_id) {
+            lv_opengles_window_texture_remove(texture);
+        }
+        else {
+            lv_display_delete(texture->disp);
+        }
+    }
+
+    lv_ll_remove(&egl_window_ll, window);
+    lv_free(window);
+
+#if !LV_USE_DRAW_OPENGLES
+    if(lv_ll_is_empty(&egl_window_ll)) {
+        GL_CALL(glDeleteTextures(1, &window_display_texture));
+        window_display_texture = 0;
+    }
+#endif
+}
+
+lv_opengles_window_texture_t * lv_opengles_window_add_texture(lv_opengles_window_t * window, unsigned int texture_id,
+                                                              int32_t w, int32_t h)
+{
+    lv_opengles_window_texture_t * texture = lv_ll_ins_tail(&window->textures);
+    LV_ASSERT_MALLOC(texture);
+    if(texture == NULL) return NULL;
+    lv_memzero(texture, sizeof(*texture));
+    texture->window = window;
+    texture->texture_id = texture_id;
+    texture->disp = lv_opengles_texture_get_from_texture_id(texture_id);
+    lv_area_set(&texture->area, 0, 0, w - 1, h - 1);
+    texture->opa = LV_OPA_COVER;
+
+#if LV_USE_DRAW_OPENGLES
+    window->direct_render_invalidated = 1;
+#endif
+
+    return texture;
+}
+
+void lv_opengles_window_texture_remove(lv_opengles_window_texture_t * texture)
+{
+    if(texture->texture_id == 0) {
+        LV_LOG_WARN("window displays should be deleted with `lv_display_delete`");
+        return;
+    }
+
+#if LV_USE_DRAW_OPENGLES
+    texture->window->direct_render_invalidated = 1;
+#endif
+
+    lv_ll_remove(&texture->window->textures, texture);
+    lv_free(texture);
+}
+
+lv_display_t * lv_opengles_window_display_create(lv_opengles_window_t * window, int32_t w, int32_t h)
+{
+    lv_display_t * disp = lv_display_create(w, h);
+    if(disp == NULL) {
+        return NULL;
+    }
+
+    lv_opengles_window_texture_t * dsc = lv_ll_ins_tail(&window->textures);
+    LV_ASSERT_MALLOC(dsc);
+    if(dsc == NULL) {
+        lv_display_delete(disp);
+        return NULL;
+    }
+    lv_memzero(dsc, sizeof(*dsc));
+    dsc->window = window;
+    dsc->disp = disp;
+    lv_area_set(&dsc->area, 0, 0, w - 1, h - 1);
+    dsc->opa = LV_OPA_COVER;
+
+#if LV_USE_DRAW_OPENGLES
+    static uint8_t LV_ATTRIBUTE_MEM_ALIGN dummy_buf;
```

Comment:
```suggestion
    static size_t LV_ATTRIBUTE_MEM_ALIGN dummy_buf;
```

I know this is also done in other drivers, but this is causing an assertion error for me inside `lv_display_set_buffers` 

```c
    LV_ASSERT_FORMAT_MSG(buf1 == lv_draw_buf_align(buf1, cf), "buf1 is not aligned: %p", buf1);
```

Using `size_t` ensures it will always be aligned